### PR TITLE
Add grammar for Gemfile.lock files

### PR DESCRIPTION
**Pull request by hmarr**
## Description

I noticed that `Gemfile.lock` files are currently treated as Ruby code (which they're not), which means the syntax highlighting isn't right. [Here's an example](https://github.com/rails/rails/blob/main/Gemfile.lock).

Given the prevalence of `Gemfile.lock` files (GitHub search [reports > 1M](https://github.com/search?q=filename%3AGemfile.lock)) I thought it'd be worth fixing. This PR adds a [TextMate grammar](https://github.com/hmarr/gemfile-lock-tmlanguage) for `Gemfile.lock` files so we can highlight them correctly.

I see this has been [discussed previously](https://github.com/github/linguist/issues/1740), but that issue went stale and was closed. If you'd rather not include a grammar for a generated file format, no hard feelings! But in that case it'd be good to remove `Gemfile.lock` from the file types classed as Ruby, and fix the misclassification problem described in the issue I linked 😄 

Right now this grammar is in its own repo as that was the path of least resistance. However I'd be more than happy to contribute it to [atom/language-ruby](https://github.com/atom/language-ruby) if that'd be a better place for it.

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?q=filename%3AGemfile.lock&type=Code _(note: as this grammar works only for files with a specific name, I changed "extension" to "filename" in the search query)_
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/rails/rails/blob/main/Gemfile.lock
    - Sample license(s):
      - https://github.com/rails/rails/blob/main/MIT-LICENSE
  
  - [x] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgithub.com%2Fhmarr%2Fgemfile-lock-tmlanguage%2Fblob%2Fmain%2Fsyntaxes%2Fgemfile-lock.tmLanguage.json&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Frails%2Frails%2Fblob%2Fmain%2FGemfile.lock&code=
  - [x] I have updated the heuristics to distinguish my language from others using the same extension. _(I don't think this is necessary as it doesn't match any extensions, but let me know if I've misunderstood!)_

### CODE REVIEW

**Comment by hmarr** [comment]
I'm not sure what to do about this test failure:

> Ruby Gemfile.lock has a sample with extension (.lock) that isn't explicitly defined in languages.yml

Adding `.lock` to `extensions` in the `languages.yml` file doesn't feel like the right solution – I assume that'd mean all `.lock` files get associated with this grammar?

**Comment by Nixinova** [comment]
> I'm not sure what to do about this test failure:

The sample file must be placed in a "filenames" subfolder.

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -4995,6 +4994,15 @@ Ruby:
   - Vagrantfile
   - buildfile
   language_id: 326
+Ruby Gemfile.lock:
+  type: data
+  color: "#701516"
+  searchable: false
```
Comment (Nixinova) [inline lib/linguist/languages.yml:5000]: @lildude: What does `searchable` do? It's on e.g. JSON but github.com/search?q=language:JSON&type=code works.

**Comment by hmarr** [comment]
Thanks for the tip @Nixinova, that seems to have done the job. ✨

File: lib/linguist/languages.yml (hunk 1 above)
Comment (lildude) [reply lib/linguist/languages.yml:5000]: https://github.com/github/linguist/issues/5269

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -4995,6 +4994,15 @@ Ruby:
   - Vagrantfile
   - buildfile
   language_id: 326
+Ruby Gemfile.lock:
```
Comment (Alhadis) [inline lib/linguist/languages.yml:4997]: There's no other type of `Gemfile.lock` in existence other than Ruby's, so I'd rename this to

```suggestion
Gemfile.lock:
```

**Review by Alhadis — CHANGES_REQUESTED** [review]

File: lib/linguist/languages.yml (hunk 2 above)
Comment (hmarr) [reply lib/linguist/languages.yml]: 👍  done

**Review by Alhadis — APPROVED** [review]

**Review by lildude — APPROVED** [review]

**Pull request merged**